### ISSUE TITLE
Trouble with creating a HA Vault cluster

### ISSUE THREAD

**Issue by osadalakmal** [hashicorp/vault#2727]
I am trying to bring up a HA Vault cluster on top of docker. The cluster is to be backed by a consul cluster. The vault cluster is one active node and two stand by nodes. The docker compose file for the setup is [here](https://gist.github.com/osadalakmal/8c90bad5a8d15cd0d48fcd8dc69b3333). 

This setup can currently bring up a consul cluster correctly and the three vault instances correctly. But the vault instances dont seem to know about each other. These are the two results of `vault status` commands on two vault instances.

**vault_alpha_1**
```
Sealed: false
Key Shares: 5
Key Threshold: 3
Unseal Progress: 0
Unseal Nonce:
Version: 0.7.0
Cluster Name: test_vault_cluster
Cluster ID: 02b772cc-16dd-a6cb-60bb-087df9a7640e

High-Availability Enabled: true
	Mode: active
	Leader: http://127.0.0.1:8200
```

**vault_bravo_1**
```
Sealed: false
Key Shares: 5
Key Threshold: 3
Unseal Progress: 0
Unseal Nonce:
Version: 0.7.0
Cluster Name: test_vault_cluster
Cluster ID: 02b772cc-16dd-a6cb-60bb-087df9a7640e

High-Availability Enabled: true
	Mode: standby
	Leader: http://127.0.0.1:8200
```

**Comment by chrishoffman** [comment]
Can you please provide your vault configs?

**Comment by osadalakmal** [comment]
```
backend "consul" {
  address = "consul_alpha:8500"
  path = "vault"
  scheme = "http"
}

cluster_name = "test_vault_cluster"

listener "tcp" {
  address = "0.0.0.0:8200"
  tls_disable = 1
}

disable_mlock = true
```

**Comment by raoofm** [comment]
@osadalakmal I think you are missing cluster_addr and redirect_addr in the backend configuration.

Also your listener config should have address and cluster_addr (optional) which is advertised for other members and it cannot be localhost or 127.0.0.1

Are you sure the the address is overwritten. if yes then make sure it is the correct hostname/ip/dns name

**Comment by osadalakmal** [comment]
@raoofm : Thanks, that seems to have fixed it. Providing redirect address on the env through docker compose file fixed it. I had to do some jugglery that was related to #1147 but got there in the end.

If anyone is interested, I was tested by directly running vault command line on the same hosts that ran vault server. This meant that `VAULT_ADDR` was already set and so was `VAULT_REDIRECT_ADDR`. Since I had set redirect addr to be https but had tls_disabled I had problems.
